iml-services/iml-device/src/lib.rs

```diff
@@ -469,35 +481,69 @@ pub fn build_updates(x: Changes<'_, Target>) -> Vec<Target> {
 
 fn parse_filesystem_data(query_result: Option<Vec<Node>>, tag: &str) -> TargetFsRecord {
     let target_to_fs = if let Some(nodes) = query_result {
-        nodes
+        let items = nodes
             .into_iter()
             .filter_map(|x| x.series)
             .map(|xs| {
                 xs.into_iter()
                     .map(|x| {
-                        x.columns
+                        let columns = x.columns;
+                        x.values
                             .into_iter()
-                            .zip(x.values.into_iter().flatten())
-                            .collect::<HashMap<String, serde_json::Value>>()
+                            .map(|v| {
+                                columns
+                                    .iter()
+                                    .cloned()
+                                    .zip(v.into_iter())
+                                    .collect::<HashMap<String, serde_json::Value>>()
```

Comment:
I get the feeling that we should create a `From` impl that takes in a tuple of (columns, values) and produces a https://docs.serde.rs/serde_json/value/enum.Value.html#variant.Object. 

Once we have an `Object` we can then deserialize it directly to a struct instead of having to do the check / panics we are doing here.

Reply:
I took a stab at this in the following patch. If this isn't quite what you're looking for can we discuss further?

Reply:
I had something slightly different in mind. In the interest of time, I'll push what I was thinking and we can discuss if you'd like.

Reply:
Pushed